### ISSUE TITLE
global variables not found?

### ISSUE THREAD

**Issue by randy3k** [jimhester/lintr#27]
![screen shot 2014-12-29 at 12 17 19 am](https://cloud.githubusercontent.com/assets/1690993/5567055/13551b4e-8ef0-11e4-9523-a21b6dd0fdf2.png)

Script global variables should be detected, but they are not.

```
SublimeLinter: r output:
/var/folders/w9/7w10svxj5093z4831n7m27w00000gp/T/SublimeLinter3-randy/untitled.lintr:8:5: warning: no visible global function definition for ‘foo’, Did you mean 'for'?
    foo(y) + increment
    ^~~
/var/folders/w9/7w10svxj5093z4831n7m27w00000gp/T/SublimeLinter3-randy/untitled.lintr:8:14: warning: no visible binding for global variable ‘increment’
    foo(y) + increment
             ^~~~~~~~~ 
```

**Comment by jimhester** [comment]
Need to tweak logic at [R/object_usage_linter.R#L8-L19](https://github.com/jimhester/lintr/blob/0378b0b9fc9b3a74256bcd6cbfc6abc8a9c0f00c/R/object_usage_linter.R#L8-L19) to evaluated the current file and include the top level names.

**Comment by KDr2** [comment]
Is someone working on this? or I can disable some linter to avoid this? 
Thanks

**Comment by jimhester** [comment]
You can disable the object_usage_linter to avoid it.

**Comment by galadude** [comment]
I'm having this problem aswell. Can you tell me how to disable object_usage_linter?

**Comment by user01** [comment]
@galadude You can set the used linters when calling lint. For example:

`lintr::lint_package(linters=lintr::default_linters[-14])`

I checked `names(lintr::default_linters)`, "found object_usage_linter" index, and removed it for the lint.

**Comment by fangly** [comment]
I am also having "no visible global function definition for 'bakName'" and "no visible binding for global variable for global variable 'test'" lints, though, they only happen in my "tests/testthat/helper_utils.R" file:
```R
test <- new.env()


bakName <- function(funName, envir) {
  paste0(environmentName(envir), "_", funName, ".bak")
}

overwriteFunction <- function(oldFun, newFun, envir=baseenv()) {
  assign(bakName(oldFun, envir), get0(oldFun, envir = envir), envir = test)
  assign(oldFun, newFun, envir = envir)
}
```
 My "R/\*" files do not generate it, and neither do my "tests/testthat/test*.R" files.

**Comment by bcipolli** [comment]
Any sense of priority on this? The linter complains about functions defined within the same file, not being found; this seems pretty basic.

Disabling the linting check ... defeats the purpose of having a linter. It's a critical check for a linter to do.

It's been almost 3 years; can anything be done to bump up the priority on this? I'm not sure my R chops are up to snuff to tackle this one..

**Comment by leonelgalan** [comment]
I'm very new to R, so I'm not sure if this is the same issue but doing:

```R
library(bar)
foo <- bar()
```

Give me a "no visible global function for `bar`, did you mean 'x'?" This code works, this library, defines the function, is just a **lintr** issue.

Is the key still in https://github.com/jimhester/lintr/blob/master/R/object_usage_linter.R#L8-L19 like suggested by @jimhester? He said "to evaluated the current file and include the top level names.", any one has pointers of where to get started?

**Comment by tsikerdekis** [comment]
An update on the exclusion strategy. It is better to remove by name than by index position:

`lintr::default_linters <- subset(lintr::default_linters,!lintr::default_linters %in% lintr::default_linters["object_usage_linter"] )`

**Comment by mnist91** [comment]
I still encounter this issue when running lint-checks on github workfows. The very own functions of the package are not found by lintr. Has anyone solved this issue rather than disabling the object_usage_linter?

**Comment by MichaelChirico** [comment]
Hi @mnist91, please open a new issue with an example if you're able.

FYI, search around other issues with the same 'object linters' tag first, it might solve your issue. if so, still feel free to open a documentation  request if you think that can be improved. Thanks!

**Comment by AshesITR** [comment]
Hi @alicebyers5, if you are developing a package, installing the package prior to linting will eliminate all object usage warnings related to symbols in the package namespace.
There is a [documentation issue](https://github.com/r-lib/lintr/issues/1137) to track alerting users of this quirk.

**Comment by allefeld** [comment]
The same happens with variables which only exist within a data frame, within pipelines using e.g. `mutate` or `pivot_wider`.

**Comment by zachmayer** [comment]
> Hi @alicebyers5, if you are developing a package, installing the package prior to linting will eliminate all object usage warnings related to symbols in the package namespace. There is a [documentation issue](https://github.com/r-lib/lintr/issues/1137) to track alerting users of this quirk.

Installing the package prior to linting fixed this for me, thank you!
